src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/FindCommandParser.java

```diff
@@ -122,11 +115,12 @@ public FindCommand parse(String args) throws ParseException {
 
         if (argMultimap.getValue(PREFIX_TITLE).isPresent()) {
             List<String> titleKeyphrases = argMultimap.getAllValues(PREFIX_TITLE);
-            try {
-                titleKeyphrases.forEach(s -> StringUtil.checkEmptyString(s));
-            } catch (Exception e) {
-                throw new ParseException(
-                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_EMPTY_TITLE_KEYPHRASE));
+            boolean hasEmptyTitleKeyphrase = false;
+            for (String s: titleKeyphrases) {
+                hasEmptyTitleKeyphrase = isEmptyString(s);
+            }
```

Comment:
Should there be a `break` statement? Otherwise, `hasEmptyTitleKeyphrase` merely depends on the last element in `titleKeyphrases`.